kubernetes/resource_kubernetes_deployment_test.go

```diff
@@ -2825,3 +2688,87 @@ func testAccKubernetesDeploymentConfigWithResourceFieldSelector(rcName, imageNam
 }
 `, rcName, imageName, resourceName, divisor)
 }
+
+func testAccKubernetesDeploymentConfig_beforeUpdate(name, imageName string) string {
```

Comment:
This is fine for now but it occurred to me that we might want to explicitly version these tests by calling them `v1` and `v2` rather than `beforeUpdate` and `afterUpdate`. It's foreseeable that we could have additional state migrations on this resource.